src/movies.js

```diff
@@ -1,16 +1,83 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function getAllDirectors(movieArray) {
 
-// Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
-// Iteration 4: Drama movies - Get the average of Drama Movies
-
-// Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-
-// Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-
-// BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-
-// BONUS - Iteration 8: Best yearly rate average - Best yearly rate average
+    const directors = movieArray.map(function(movieArray) {
+        return movieArray.director;
+    })
+        return directors;
+    }
+    
+    getAllDirectors(movies);
+    console.log(getAllDirectors(movies))
+    
+    // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+    
+    function howManyMovies(movieArray) {
+        
+     let amountOfMovies = movieArray.filter(function(movieArray){
+        if (movieArray.director === "Steven Spielberg" && movieArray.genre.includes("Drama")) {
+        return true
+        }
+        })
+    return amountOfMovies.length
+    }
+        
+    console.log(howManyMovies(movies));
+    
+    
+    // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+    
+    function ratesAverage(movieArray) {
+    
+    let sum = movieArray.reduce(function(rate, movieArray) {
+        return rate + movieArray.rate
+     }, 0)
+     return (sum / movieArray.length).toFixed(2);
+    }
+    
+    ratesAverage(movies);
+    
+    
+    
+    
+    // Iteration 4: Drama movies - Get the average of Drama Movies
+    
+    function dramaMoviesRate(movieArray) {
+    
+        let sumDrama = movieArray.reduce(function(rate, movieArray) {
+            if (movieArray.genre.includes("Drama")) {
+            return rate + movieArray.rate
+            }
+         }, 0)
+         return (sumDrama / movieArray.length).toFixed(2);
+        }
+        
+     dramaMoviesRate(movies);
+    
+    // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+    
+    function orderByYear(movieArray) {
+        let sortedMovies = movieArray.year.sort(function(a, b) {
+        return a - b
+        })
+        return sortedMovies
+    }
+    
+    orderByYear(movies);
+    
+    // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
+    
+    function orderAlphabetically(movieArray) {
+        let sortedByTitle = movieArray.title.sort(function(a, b) {
```

Comment:
inside your sort method, you need to use your parameters a and b. That's how sort knows how to sort.